apps/src/lab2/responseValidators.ts

```diff
@@ -38,9 +38,9 @@ const PythonSourceResponseValidator: ResponseValidator<
     if (typeof responseToValidate.source === 'string') {
       throw new ValidationError('Python sources must be a JSON object');
     }
-    const source = responseToValidate.source as NestedSourceCode;
+    const source = responseToValidate.source as MultiFileSource;
     // TODO: support a nested main.py
-    if (!source['main.py']) {
+    if (!source?.files?.['main.py']) {
```

Comment:
fyi @thomasoniii for Python Lab I kept the keys of the file map as the file names rather than file ids. This made it easier to look up the main file. We can discuss if we want to do this for all labs or do ids for all labs.

Reply:
I think I went with ids for two reasons (I think I originally had names as well). 

The main one was just renaming the file - My structure was keying off of name _and_ storing the name as an attribute, so a name change also meant changing the key pointing to the object. Not the end of the world keeping the two things in sync, but since the id should be immutable it doesn't pop up for that case since the user can't change it.

And the other was consistency with the folder approach. Since I'd set it up so that the file object points to the folder object (as opposed to the folder containing a list of files), I'd gone with `folderId` because otherwise if the folder was renamed you'd have to go update the `folderName` attributes in all of the related files.

Note that if the data was stored the other way - where a folder contains a list of fileIds, then we'd definitely want to key off of `fileId` instead of `fileName` for the same reason - if you renamed a file, you'd also need to go update that folder's list of files to include the new name.

Anyway, to reduce the redundant updates at least one of the objects really needed an ID value, so I finally just went with including it on both of them for consistency's sake.

All that said, it's certainly open to debate and we can adjust and re-tool as desired.

Reply:
That makes sense. I'll update python to use ids for file keys.